rust_src/src/frames.rs

```diff
@@ -593,4 +593,23 @@ pub fn frame_focus(frame: LispFrameLiveOrSelected) -> LispObject {
     frame_ref.focus_frame
 }
 
+/// Set the input focus to FRAME.
+/// FRAME nil means use the selected frame.  Optional argument NOACTIVATE
+/// means do not activate FRAME.
+///
+/// If there is no window system support, this function does nothing.
+#[lisp_fn(min = "1")]
+pub fn x_focus_frame(_frame: LispFrameLiveOrSelected, _noactivate: LispObject) -> LispObject {
```

Comment:
You can set the type of `noactivate` to bool, that way you can pass it directly to `c_x_focus_frame`.